crates/nu-std/CONTRIBUTING.md

```diff
@@ -0,0 +1,191 @@
+
+    ```
+    cd /path/to/nushell/repo
+    code .  # opinionated: we love vscode!
+    (open folder crates/nu-std in IDE)
+   ```
+### The PR
+If your idea is mostly a bug fix, you might not open a PR till you've got a fix (nearly)ready to go.  
+
+But if you're considering new commands or substantial or (even simple) breaking changes, 
+it's good to advertise your proposed work to other collaborators for their awareness 
+and to solicit their early feedback. 
+Often, this feedback will result in a better and more maintainable solution long term, 
+so give your fellow collaborators the opportunity.  Someday, they'll return the favor...
+
+To open a PR before you've got code ready to go:
+  1. create a feature branch in your fork of the repo
+  2. make a trivial code change and push it to Github
+  4. open a PR on the standard library repo
+  5. **mark it as a work in progress (WIP)**  
+This signals that you are not yet asking to have the PR formally reviewed and merged.
+  6. provide a preliminary description of your proposed change.     
+Your description should include the external interface for your feature, 
+a draft of the command arguments and signature, 
+new/changed environment variables and any other user-visible changes.
+
+### Design considerations
+The standard library consists mostly of Nushell custom commands and their associated environment variables.  
+For background on these Nushell features, see 
+[Modules chapter of the Nushell book](https://www.nushell.sh/book/modules.html). 
+Existing code in the standard library provides pretty good examples to work from.  
+
+1. Create / update the source and test files for a custom command `foo` as follows:
+   ```shell
+   # source file 
+   /path/to/nushell/crates/nu-std/lib/foo.nu  
+   # corresponding unit test file
+   /path/to/nushell/crates/nu-std/tests/test_foo.nu
+   ```
+   Thou shalt provide unit tests to cover your changes.   
+2. Typically, the source file will implement multiple subcommands and possibly a main command as well.  
+   For an example of a custom command which does not modify environment variables (but may reference them), see:
+   ```shell
+   /path/to/nushell/crates/nu-std/lib/assert.nu
+   ```
+   Note the use of `export def` to define the subcommand.  
+
+   For an example of a custom command which *does* modify environment variables, see:
+   ```shell
+   /path/to/nushell/crates/nu-std/lib/dirs.nu
+   ```
+   Note the use of `export def-env` to define the subcommand, 
+   the use of `export-env {}` to initialize the environment variable and  `let-env` to update it. 
+
+1. A `foo` command will be exposed to the user as `std foo` (at a minimum).  
+To enable this, update file `/path/to/nushell/crates/nu-std/lib/mod.rs` and add this code:
+   ```
+   export use crates/nu-std/lib/foo.nu *    # command doesn't update environment
+   export-env {
+          use crates/nu-std/lib/bar.nu *    # command *does* update environment
+   }
```

Comment:
this is actually incorrect but it's a recent change in #8815 :relieved: 

should be
```
   export use foo *    # command doesn't update environment
   export-env {
          use bar *    # command *does* update environment
   }
```

:bulb: modules are automatically loaded from `crates/nu-std/lib/` as `.nu` files thanks to the library loader in `crates/nu-std/src/lib.rs` :+1:

Reply:
OK, the advice is "monkey see, monkey do", but it means std is becoming less and less like a typical user-written module library.

Reply:
ehe nope it's not anymore :yum: 

we decided, for now at least, to embed the library in the `nu` binary directly => the direct consequence is that we can not rely on paths like `crates/nu-std/lib/<mod>.nu` because the user won't have that file at all on the system!
only specifying `<mod>` is all the automatic loader (in `crates/nu-std/src/lib.rs`) needs to know :+1: